File: 1768-merge-strings-alternately/1768-merge-strings-alternately.cpp
class Solution {
public:
    string mergeAlternately(string word1, string word2) {
        string s = "";
        
        int w1 = 0, w2 = 0;
        
        while (w1 < word1.size() && w2 < word2.size()) {
            s.push_back(word1[w1++]);
            s.push_back(word2[w2++]);
        }
        
        while (w1 < word1.size()) {
            s.push_back(word1[w1++]);
        }
        
        while (w2 < word2.size()) {
            s.push_back(word2[w2++]);
        }
        
        return s;    
    }
};

File: 1768-merge-strings-alternately/1768-merge-strings-alternately.rs
impl Solution {
    pub fn merge_alternately(word1: String, word2: String) -> String {
        let s1: Vec<char> = word1.chars().collect();
        let s2: Vec<char> = word2.chars().collect();
        
        let n = word1.len().max(word2.len());
        let mut i = 0;
        
        let mut res = String::new();
        
        while i<n {
            if(i<word1.len()) {res.push(s1[i]);}
            if(i<word2.len()) {res.push(s2[i]);}
            i += 1;
        }
        
        res
    }
}